core/src/main/java/hudson/console/ModelHyperlinkNote.java

```diff
@@ -57,6 +57,10 @@ public static String encodeTo(Node node) {
     }
 
     public static String encodeTo(String url, String text) {
+        // If text contains newlines, then its stored length will not match its length when being
+        // displayed, since the display length will only include text up to the first newline,
+        // which will cause an IndexOutOfBoundsException in MarkupText#rangeCheck when viewing the note.
+        text = text.replace('\n', ' ');
```

Comment:
Might need to replace carriage returns as well, I need to investigate.

Reply:
I as well would say it's necessary. If the text has \n\r and the method replaces only the \n, the String could become weird somehow.

Reply:
The code that reads the text with console notes and converts it into markup [uses `\n` as the end-of-line marker](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/console/LineTransformationOutputStream.java#L112), and `\r` is not treated specially, so it shouldn't affect the ConsoleNote to markup conversion.

I don't think a bare `\r` in the output should cause anything to be displayed strangely (at least not more strangely than it would have before this fix), but we could also replace it with a space if we want to be extra careful.